pkg/icingadb/schema.go

```diff
@@ -15,27 +18,45 @@ const (
 	expectedPostgresSchemaVersion = 4
 )
 
-// CheckSchema asserts the database schema of the expected version being present.
+// ErrSchemaNotExists implies that no Icinga DB schema has been imported.
+var ErrSchemaNotExists = stderrors.New("no database schema exists")
+
+// ErrSchemaMismatch implies an unexpected schema version, most likely after Icinga DB was updated but the database
+// missed the schema upgrade.
+var ErrSchemaMismatch = stderrors.New("unexpected database schema version")
+
+// CheckSchema verifies the correct database schema is present.
+//
+// This function returns the following error types, possibly wrapped:
+//   - If no schema exists, the error returned is ErrSchemaNotExists.
+//   - If the schema version does not match the expected version, the error returned is ErrSchemaMismatch.
+//   - Otherwise, the original error is returned, e.g., for general database problems.
```

Comment:
I don't think the comma after e.g. is necessary.

Reply:
Usually a comma follows after "e.g.", but not "i.e.".

https://english.stackexchange.com/questions/16172/should-i-always-use-a-comma-after-e-g-or-i-e